src/services/pcn-packetcapture/src/Packetcapture_dp_ingress.c

```diff
@@ -70,128 +53,15 @@ BPF_ARRAY(packet_timestamp, uint64_t, 1);
  */
 BPF_ARRAY(working, uint8_t, 1);
```

Comment:
where are you using this BPF_ARRAY?
Are you able to differentiate the ebpf code that is activated? For example, can we capture only outgoing traffic and not incoming traffic?

Reply:
I can capture only outgoing or incoming or both because I inject the filterCode according to capture direction inserted by the user, if I capture only in one way in the other way the filter is "RETURN RX_OK;"

if (CapStatus == 3 || CapStatus == 2) {
    reload(codeINGRESS,0,ProgramType::INGRESS);
    reload(codeEGRESS,0,ProgramType::EGRESS);
  } else {
      if (CapStatus == 0) {
        reload(codeINGRESS,0,ProgramType::INGRESS);
      } else {
          if (CapStatus == 1) {
            reload(codeEGRESS,0,ProgramType::EGRESS);
          }
      }
  }

Reply:
Well, you should then delete all references to using "working" maps if you no longer use them. Look for example in the fast paths of the service (ingress and egress) and in packetcapture.cpp where they are updated. you should also delete the definition of the associated enum

Reply:
Ok perfect done, thanks